src/parser.cpp

```diff
@@ -11,6 +11,22 @@ namespace vpd
 Parser::Parser(const std::string& vpdFilePath, nlohmann::json parsedJson) :
     m_vpdFilePath(vpdFilePath), m_parsedJson(parsedJson)
 {
+    std::error_code l_errCode;
+    auto l_filesize = -1;
+
+    if (!std::filesystem::exists(m_vpdFilePath, l_errCode) ||
+        ((l_filesize = std::filesystem::file_size(m_vpdFilePath, l_errCode)) ==
+         0))
+    {
+        const std::string l_message{
```

Comment:
Why such complex logging?
If "file_size" call fails then "l_filesize" will always be -1. What is the point printing it?

suggesstion:
in the body, just check for 
if(l_errCode)
{
throw the message, //create a message that parser creation failed for reason l_errCode.message?
}

Reply:
will get inside the if case, if one of the checks fails.
As @jinuthomas suggested to use -1, in case of first condition fails will have -1 as size in log along with file path & error message if exists.

Reply:
Hmm, I am a bit confused here.

Let's say first condition fails then we don't have a valid file size extracted at this point, why do we need to assume that file size is -1 and print it. Wouldn't it will be a mis-representation?

Now, If the second condition fails then it's implicit that file size returned will always be -1 in that case.

Which one failed can be known by just printing .message().

Parser creation failed with reason l_errCode.message() should suffice.

As it is just a log, I don't have a strong push here.
Take a call.

Reply:
should this also contain the check for the json empty ?

Reply:
if json is empty, we still continue to execute further as zero offset.